json-simple/src/main/java/org/json/simple/ItemList.java

```diff
@@ -40,7 +40,18 @@ public List getItems(){
 	}
 	
 	public String[] getArray(){
-		return (String[])this.items.toArray();
+		List<String> stringItems = new ArrayList<String>();
+		for(Object object : this.items) {
+			if(object instanceof String) {
+				stringItems.add((String)object);
+			}
+		}
+		String[] items = new String[stringItems.size()];
+		for(int i = 0; i < stringItems.size(); i++) {
+			items[i] = stringItems.get(i);
+		}
+		
+		return items;
```

Comment:
leaving a cool comment on the review part. the first for cycle'd look even cooler, if it used lambda expressions.